website/blog/2022-01-21-components-and-blocks.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Components and blocks
+description: Explanation of the difference between components and blocks on a practical example
+slug: components-and-blocks
+authors: iobrado
+tags: [eightshift, boilerplate, components, blocks]
+image: #
+hide_table_of_contents: false
+---
+In this blog post, we'll give you a practical explanation about the difference between components and blocks. For a truly in-depth explanation about components and blocks, you can refer to [our documentation](https://infinum.github.io/eightshift-docs/docs/basics/blocks). A more specific explanation about component in a block can be found in [Component in a Block](https://infinum.github.io/eightshift-docs/docs/basics/blocks-component-in-block/) section.
+
+<!--truncate-->
+One of the things that is really confusing to newcomers is the difference between components and blocks. Let's break it all down.
+
+### Component
+
+**A component is like a blank slate, a template without context**. It doesn't know or even cares for what it will be used. Can you combine multiple simple components into one complex component? Absolutely! That's exactly what we'll do in the following example. 
+
+Let's say we have a component that consists of an image and a heading. Image and heading are two simple components that come with the theme out-of-the-box.
+
+Here are a few examples of how this new component can be used:
+- displaying a post
+- displaying a user or an author
+- displaying a photo with the photographer's name
+- and many more...
+
+As you can already see, there are many ways you can use this component. What helps even more to expand the usability of this component are `attributes`. Let's say we have the following attributes:
+- `imagePosition` - left, right, top, bottom
+- `imageShape` - rectangular, circular
+- `imageWidth` - 25%, 50%, 75%
+- `headingColor` - black, blue, red
+- `headingSize` - small, medium, large
+
+These attributes give you the option to modify the look of the component, so this specific component can be used for the following:
+
+- **related posts** - rectangular image on the left side, 25% image width, heading takes up the remaining 75%, medium heading colored in blue
+- **user avatar** - circular image on the left side, 25% image width, small heading colored in black
+- **photo of the day** - rectangular image on the right side which takes 75% width and photographer's name is written in medium blue text
+- **featured posts** - rectangular image on top, large heading in red color
+
+You even have the option to add conditional logic, e.g. to have the option for `imageWidth` only when `imagePosition` is set to left or right. And when `imagePosition` is set to top or bottom, it defaults to 100% width.
```

Comment:
Not sure what you're trying to say here - that you can conditionally hide or show options in Gutenberg?